### ISSUE TITLE
Wrong lemmatization in v2 (POS not used)

### ISSUE THREAD

**Issue by thomasopsomer** [explosion/spaCy#1548]
It seems that lemmatization isn't working correctly in v2:
```python
import spacy
nlp = spacy.load("en")

doc = nlp("machine learning is a field of computer science")
doc[1].pos_   # 'NOUN'
doc[1].lemma_   # 'learn' :/
```
Whereas is 1.9, I get the correct lemma: "learning".

## Your Environment
* **spaCy version:** 2.0.2
* **Platform:** Darwin-16.7.0-x86_64-i386-64bit
* **Python version:** 3.6.2
* **Models:** en, en_core_web_lg, en_vectors_web_lg

**Comment by ines** [comment]
Ah damn, I think this might be caused by a lemmatization fix introduced in v2.0 which uses the lookup tables *plus* the lemmatization rules. I think what's happening here is that the lemma set by the `Lemmatizer` is overwritten by lookup table – or not overwritten correctly by the lemma rules.

**Comment by thomasopsomer** [comment]
@ines the problem seems to sit here in the [set_annotations method](https://github.com/explosion/spaCy/blob/master/spacy/pipeline.pyx#L412):
```python
for j, tag_id in enumerate(doc_tag_ids):
    # Don't clobber preset POS tags
    if doc.c[j].tag == 0 and doc.c[j].pos == 0:
        # Don't clobber preset lemmas
        lemma = doc.c[j].lemma
        vocab.morphology.assign_tag_id(&doc.c[j], tag_id)
        if lemma != 0 and lemma != doc.c[j].lex.orth:
            doc.c[j].lemma = lemma
    idx += 1
```
`vocab.morphology.assign_tag_id` sets the right lemma on the token according to its tag. But then the condition `if lemma != 0 and lemma != doc.c[j].lex.orth:` set it back to the preset lemma. I don't really get the intent of this condition ^^

**Comment by ines** [comment]
Thanks for getting to the bottom of this. Just had a look at the blame log and it looks like my initial suspicion was right then... The condition was added only 6 days ago (see 75e1618ec32a5a06c78f5737e11aa869f60d3b20 and 31babe3c3f7b02a76b817d00fa78e2583a229659) and I think the original intent was to prevent lemmas set by the *tokenizer exceptions* from being overwritten. This was something we noticed in the final tests for v2.0... but unfortunately, we didn't test property for the other cases 😞 Sorry about that!

I'll add a `help wanted` label in case someone from the community wants to help out – otherwise, we'll just fix it for the next release.
